### DISCUSSION TITLE
Protein inference

### DISCUSSION THREAD

**Discussion by NicoDrou** [vdemichev/DiaNN#830] [General]
dear Vadim,

I would have a question about the mchanism of the heuristic inference.
I understood it creates the protein group and assigns the intensity of the share peptides to the most likely protein (Protein.Group column) from the group (in the Protein.Ids ) ensity of the most likely protein from the intensity of the share peptides.
How this protein is selected?
I have here an exemple: 
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/Droui001/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/Droui001/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

Protein.Group | Protein.Ids | Protein.Names | Genes | First.Protein.Description | Proteotypic | Stripped.Sequence
-- | -- | -- | -- | -- | -- | --
P01715 | P01715 | LV301_HUMAN | IGLV3-1 | Immunoglobulin lambda variable 3-1 | 1 | PGQSPVLVIYQDSK
P01715 | P01715 | LV301_HUMAN | IGLV3-1 | Immunoglobulin lambda variable 3-1 | 1 | YACWYQQK
A0A075B6K4 | A0A075B6K4 | LV310_HUMAN | IGLV3-10 | Immunoglobulin lambda variable 3-10 | 1 | SGQAPVLVIYEDSK
A0A075B6K4 | P01715;A0A075B6K4 | LV310_HUMAN | IGLV3-10 | Immunoglobulin lambda variable 3-10 | 0 | RPSGIPER
A0A075B6K4 | A0A075B6K0;A0A075B6K4 | LV310_HUMAN | IGLV3-10 | Immunoglobulin lambda variable 3-10 | 0 | YAYWYQQK

</body>

</html>

The protein P01715 has 2 proteotypic peptides detected and one shared with A0A075B6K4. 
A0A075B6K4 has one proteotypic peptide, one shared with P01715 and one with A0A075B6K4.
But the shared peptide between A0A075B6K4 and P01715 is finally assigned to A0A075B6K4.

Also, when this option is disabled, DIA-NN V1.8.1 identifies more proteins in the unique.gene report. 
From my understanding, any protein with a proteotypic peptide should be alone in the group and only the shared peptides are used to create the multi-protein. But it shouldn't affect the number of proteins identified with proteotypic peptides.
Or maybe I misunderstood something?

I attach here the log files and the unique.genes reports of the comparison I made.
[Protein inference.zip](https://github.com/vdemichev/DiaNN/files/12894251/Protein.inference.zip)

Many thanks in advance for your help.
Regards,
Nicolas

**Comment by vdemichev** [comment]
Hi Nicolas,

Proteotypicity is not taken into account here, DIA-NN just counts the number of precursors matching a protein, and among proteins with equal numbers prioritises swissprot ones (unless you turn this feature off) and among the rest just prioritises proteins alphabetically.

Different numbers in unique_genes matrix - in which case does it correctly match the number of genes with valid Genes.MaxLFQ.Unique in the main report?

Best,
Vadim

**Comment by Althalis** [comment]
Hello, 
just to be sure, when we turn off the 'heuristic protein inference' (unticking the box), which protein inference method is used then?

Also, can you clarify when/why using heuristic protein inference might be a problem with differential abundance analyses? (as stated by the hover over popup text)

Thank you for your time, 
Best,
Julie

**Reply by vdemichev** [reply]
Hi Julie,

"when we turn off the 'heuristic protein inference' (unticking the box), which protein inference method is used then?" - the old method, which will include same protein ID in multiple groups.

"when/why using heuristic protein inference might be a problem with differential abundance analyses?" - If you have two proteins, A and B, with peptides A:X,Y, B:Y,Z, then heuristic will quantify A with X and Y and B with Z only. It's OKish to use (and almost everyone in proteomics does), but good to keep the limitations of protein inference in mind.

Best,
Vadim
